src/android/Fingerprint.java

```diff
@@ -1,454 +1,296 @@
+                        }
+                        if(arg_object.optString("fallbackButtonTitle") != null && !arg_object.optString("fallbackButtonTitle").isEmpty()){
+                            mFallbackButtonTitle = arg_object.getString("fallbackButtonTitle");
+                        }else{
+                            if(!mDisableBackup){
+                                mFallbackButtonTitle = "Use Backup";
                             }
                         }
-                    });
-                    mPluginResult.setKeepCallback(true);
-                } else {
-                    mCallbackContext.sendPluginResult(mPluginResult);
-                }
+                    } catch (JSONException e) {
+
+                    }
+
+					//set up the builder
+					mBiometricManager = new BiometricManager.BiometricBuilder(mContext)
+						.setTitle(mTitle)
+						.setSubtitle(mSubtitle)
+						.setDescription(mDescription)
+						.setNegativeButtonText(mFallbackButtonTitle)
+						.build();
+
+					//start authentication
+					mBiometricManager.authenticate(Fingerprint.this);
+	            }
+	        });
+
+            PluginResult pluginResult = new PluginResult(PluginResult.Status.NO_RESULT);
+            pluginResult.setKeepCallback(true);
+            this.mCallbackContext.sendPluginResult(pluginResult);
 
-            } else {
-                mPluginResult = new PluginResult(PluginResult.Status.ERROR);
-                mCallbackContext.error(errorResponse.put("message", "Fingerprint authentication not available"));
-                mCallbackContext.sendPluginResult(mPluginResult);
-            }
             return true;
-        } else if (action.equals("isAvailable")) {
-            if(mFingerPrintManager.isHardwareDetected() && mFingerPrintManager.hasEnrolledFingerprints()){
-                mPluginResult = new PluginResult(PluginResult.Status.OK, "finger");
-                mCallbackContext.success("finger");
-            }else{
-                mPluginResult = new PluginResult(PluginResult.Status.ERROR);
-
-                if (mFingerPrintManager.isHardwareDetected() && !mFingerPrintManager.hasEnrolledFingerprints()) {
-                    mCallbackContext.error(errorResponse.put("message", "Fingerprint authentication not ready"));
-                } else {
-                    mCallbackContext.error(errorResponse.put("message", "Fingerprint authentication not available"));
-                }
+        }else if(action.equals("isAvailable")){
+
+            /**
+             * NOTE:
+             * Android 9 only includes fingerprint integration for BiometricPrompt. However, integrated
+             * support for other biometric modalities are forthcoming. So right now can't return anything
+             * other than "finger" if it's available even if face dection opens up.
+             * @link https://source.android.com/security/biometric
+             */
+            boolean passedAllChecks = true;
```

Comment:
am i missing something or does nothing set `passedAllChecks` to `false`?

Below your checking `if (!isSdkVersionSupported())` and still set `passedAllChecks = true`

Reply:
hmm... passedAllChecks I think was supposed to be removed from all IF statements and those conditions should be a series of if/else's since each failure has it's own error call back ...sendSuccess("finger") should be in the last else statement ..will test in a bit. Thanks for catching. I thins case you would received an error callback from each conditions onX event so passedAllChecks wouldn't matter. Fixing.

Reply:
Please check these syntax errors for android builds: https://travis-ci.org/NiklasMerz/cordova-plugin-fingerprint-aio/jobs/587963288#L1422

Reply:
sorry guys 🤦‍♂  been working on ts project lol

Reply:
fixed - 5e2e4b117a8348de4e34f8e536234f0560500fed

Reply:
We all know that. Hopping between languages is always fun :-D